website/content/docs/connect/config-entries/service-router.mdx

```diff
@@ -456,6 +457,15 @@ Specifies the path prefix to match on the HTTP request path. When using this fie
 - Default: None
 - Data type: String
 
+### `Routes[].Match{}.HTTP{}.CaseInsensitive`
+
+Specifies the path prefix to match on the HTTP request path must be case insensitive or not. When using this field, do not configure `PathExact` or `PathRegex` in the same HTTP map.
```

Comment:
> When using this field, do not configure `PathExact` or `PathRegex` in the same HTTP map.

I know the Envoy docs state that case insensitivity is "Ignored for safe_regex matching". Why the caution about `PathExact`?

If those are combinations we shouldn't support, ideally we'd provide some kind of validation error, but I'm not sure if that's easy to do. (There don't seem to be any validation type things in that function already. I defer to eng on this.)

Reply:
We could introduce some kind of validation whenever routes are persisted to prevent this scenario from happening. I would like a little bit more clarification on why we suggest not mixing them. @Lord-Y was there some undesirable behavior you were aware of that could occur in this scenario?

Reply:
@jkirschner-hashicorp @hashi-derek Sorry for the late answer. I put this caution because of the Envoy documentation stating `Ignored for safe_regex matching`. Maybe it's misleading. After re-reading the code here https://github.com/hashicorp/consul/blob/1a06f73575c91936fbd95feab7f79cdea14e6994/agent/xds/routes.go#L825-L846 I can discard this caution.

Reply:
I remove the extra caution here https://github.com/hashicorp/consul/pull/19647/commits/28ae6b889867c3a0c02c59826cac629efbb4f344